packages/twenty-front/src/pages/settings/data-model/SettingsObjectOverview/SettingsObjectOverview.tsx

```diff
@@ -0,0 +1,331 @@
+            ? nodeChange.positionAbsolute?.x
+            : node.position.x || 0;
+
+        incomingNodes.forEach((incomingNode) => {
+          const edge = edges.find((edge) => {
+            return edge.target === node.id && edge.source === incomingNode.id;
+          });
+
+          if (isDefined(newXPos)) {
+            setEdges((eds) =>
+              eds.map((ed) => {
+                if (isDefined(edge) && ed.id === edge.id) {
+                  const sourcePosition = calculateSourcePosition(
+                    incomingNode.width as number,
+                    incomingNode.position.x,
+                    node.width as number,
+                    newXPos,
+                  );
+                  const targetPosition = calculateTargetPosition(
+                    incomingNode.width as number,
+                    incomingNode.position.x,
+                    node.width as number,
+                    newXPos,
+                  );
+                  const sourceHandle = `${edge.data.sourceField}-${sourcePosition}`;
+                  const targetHandle = `${edge.data.targetField}-${targetPosition}`;
+                  ed.sourceHandle = sourceHandle;
+                  ed.targetHandle = targetHandle;
+                  ed.markerEnd = 'marker';
+                  ed.markerStart = 'marker';
+                }
+
+                return ed;
+              }),
+            );
+          }
+        });
+
+        const outgoingNodes = getOutgoers(node, nodes, edges);
+        outgoingNodes.forEach((targetNode) => {
+          const edge = edges.find((edge) => {
+            return edge.target === targetNode.id && edge.source === node.id;
+          });
+          if (isDefined(newXPos)) {
+            setEdges((eds) =>
+              eds.map((ed) => {
+                if (isDefined(edge) && ed.id === edge.id) {
+                  const sourcePosition = calculateSourcePosition(
+                    node.width as number,
+                    newXPos,
+                    targetNode.width as number,
+                    targetNode.position.x,
+                  );
+                  const targetPosition = calculateTargetPosition(
+                    node.width as number,
+                    newXPos,
+                    targetNode.width as number,
+                    targetNode.position.x,
+                  );
+
+                  const sourceHandle = `${edge.data.sourceField}-${sourcePosition}`;
+                  const targetHandle = `${edge.data.targetField}-${targetPosition}`;
+
+                  ed.sourceHandle = sourceHandle;
+                  ed.targetHandle = targetHandle;
+                  ed.markerEnd = 'marker';
+                  ed.markerStart = 'marker';
+                }
+
+                return ed;
+              }),
+            );
+          }
+        });
+      });
+
+      onNodesChange(nodeChanges);
+    },
+    [onNodesChange, setEdges, nodes, edges],
+  );
+
+  useEffect(() => {
```

Comment:
move this to an Effect Component
We do not allow useEffect in regular components to avoid re-renders

Reply:
It's not a big problem here but it will introduce a pattern that other contributors could follow and lead to re-rendering performance issues